src/tracer.ts

```diff
@@ -1,8 +1,48 @@
-import Span from './span';
-import SpanContext from './span_context';
 import * as Constants from './constants';
 import * as Functions from './functions';
 import * as Noop from './noop';
+import Reference from './reference';
+import Span from './span';
+import SpanContext from './span_context';
+
+export interface SpanFields {
+    /**
+     * DEPRECATED: the name to use for the newly created span. If provided,
+     * overrides the first argument to startSpan(). Provided for
+     * backwards-compatibility.
+     */
+    operationName?: string;
+
+    /**
+     * a parent SpanContext (or Span, for convenience) that the newly-started
+     * span will be the child of (per REFERENCE_CHILD_OF). If specified,
+     * `references` must be unspecified.
+     */
+    childOf?: Span | SpanContext;
+
+    /**
+     * an array of Reference instances, each pointing to a causal parent
+     * SpanContext. If specified, `fields.childOf` must be unspecified.
+     */
+    references?: Reference[];
+
+    /**
+     * set of key-value pairs which will be set as tags on the newly created
+     * Span. Ownership of the object is passed to the created span for
+     * efficiency reasons (the caller should not modify this object after
+     * calling startSpan).
+     */
+    tags?: { [key: string]: any };
+
+    /**
+     * a manually specified start time for the created Span object. The time
+     * should be specified in milliseconds as Unix timestamp. Decimal value are
+     * supported to represent time values with sub-millisecond accuracy.
+     */
+    startTime?: number;
+}
+
+export type Format = typeof Constants.FORMAT_BINARY | typeof Constants.FORMAT_TEXT_MAP | typeof Constants.FORMAT_HTTP_HEADERS;
```

Comment:
will this restrict the format argument to just these three? It is meant to be open-ended.